x-pack/legacy/plugins/ml/public/timeseriesexplorer/timeseriesexplorer.js

```diff
@@ -0,0 +1,1229 @@
+const allValuesLabel = i18n.translate('xpack.ml.timeSeriesExplorer.allPartitionValuesLabel', {
+  defaultMessage: 'all',
+});
+
+function getTimeseriesexplorerDefaultState() {
+  return {
+    chartDetails: undefined,
+    contextChartData: undefined,
+    contextForecastData: undefined,
+    // Not chartable if e.g. model plot with terms for a varp detector
+    dataNotChartable: false,
+    detectorId: undefined,
+    detectors: [],
+    entities: [],
+    focusAnnotationData: [],
+    focusChartData: undefined,
+    focusForecastData: undefined,
+    hasResults: false,
+    jobs: [],
+    // Counter to keep track of what data sets have been loaded.
+    loadCounter: 0,
+    loading: false,
+    modelPlotEnabled: false,
+    selectedJob: undefined,
+    // Toggles display of annotations in the focus chart
+    showAnnotations: mlAnnotationsEnabled,
+    showAnnotationsCheckbox: mlAnnotationsEnabled,
+    // Toggles display of forecast data in the focus chart
+    showForecast: true,
+    showForecastCheckbox: false,
+    showModelBoundsCheckbox: false,
+    svgWidth: 0,
+    tableData: undefined,
+    zoomFrom: undefined,
+    zoomTo: undefined,
+
+    // Toggles display of model bounds in the focus chart
+    showModelBounds: true,
+  };
+}
+
+const TimeSeriesExplorerPage = ({ children, jobSelectorProps, resizeRef }) => (
+  <Fragment>
+    <NavigationMenu tabId="timeseriesexplorer" />
+    <JobSelector {...jobSelectorProps} />
+    <div className="ml-time-series-explorer" ref={resizeRef} >
+      {children}
+    </div>
+  </Fragment>
+);
+
+const containerPadding = 24;
+
+export class TimeSeriesExplorer extends React.Component {
+  static propTypes = {
+    appState: PropTypes.object.isRequired,
+    dateFormatTz: PropTypes.string.isRequired,
+    globalState: PropTypes.object.isRequired,
+    mlJobSelectService: PropTypes.object.isRequired,
+    timefilter: PropTypes.object.isRequired,
+  };
+
+  state = getTimeseriesexplorerDefaultState();
+
+  subscriptions = new Subscription();
+
+  resizeRef = createRef();
+  resizeChecker = undefined;
+  resizeHandler = () => {
+    this.setState({
+      svgWidth: (this.resizeRef.current !== null) ? this.resizeRef.current.offsetWidth - containerPadding : 0,
+    });
+  }
+
+  detectorIndexChangeHandler = (e) => {
+    const id = e.target.value;
+    if (id !== undefined) {
+      this.setState({ detectorrId: id });
+    }
+    this.updateControlsForDetector();
+    this.loadEntityValues();
+  };
+
+  toggleShowAnnotationsHandler = () => {
+    if (mlAnnotationsEnabled) {
+      this.setState(prevState => ({
+        showAnnotations: !prevState.showAnnotations
+      }));
+    }
+  }
+
+  toggleShowForecastHandler = () => {
+    this.setState(prevState => ({
+      showForecast: !prevState.showForecast
+    }));
+  };
+
+  toggleShowModelBoundsHandler = () => {
+    this.setState({
+      showModelBounds: !this.state.showModelBounds,
+    });
+  }
+
+  previousTsc = {};
```

Comment:
This variable name is not very intuitive. Does it stand for 'time series chart'. I think this needs a clearer name, like the three variables below.

Reply:
Renamed to `chartProps` in 18a5945be7.